auslib/test/test_db.py

```diff
@@ -3919,11 +3915,110 @@ def testAllTablesExist(self):
             "user_roles",
             "user_roles_history",
         ])
-        self.assertEquals(set(self.db.metadata.tables.keys()), expected_tables)
+
+        # autoincrement isn't tested as Sqlite does not support this outside of INTEGER PRIMARY KEYS.
+        # If the testing db is ever switched to mysql, this should be revisited.
+        cls.properties = ('nullable',
+                          'primary_key',
+                          # 'autoincrement',
+                          'constraints',
+                          'foreign_keys',
+                          'index',
+                          'timetuple', )
+        cls.property_err_msg = ("Property '{property}' on '{table_name}.{column}' differs between model "
+                                "and migration: (model) {model_prop} != (reflected) {reflected_prop}")
+
+    def setUp(self):
+        NamedFileDatabaseMixin.setUp(self)
+        self.db = AUSDatabase(self.dburi)
+        self.db.metadata.create_all()
+        self.aus_db_tables = [self.db.rulesTable, self.db.releasesTable,
+                              self.db.permissionsTable, self.db.dockerflowTable]
+
+    def _is_column_unique(self, col_obj):
+        """
+        Check to see if a column is unique using a Sqlite-specific query.
+        """
+        col_engine = col_obj.table.metadata.bind
+        table_name = col_obj.table.name
+        res = col_engine.execute("SELECT sql FROM sqlite_master WHERE name = :table_name", table_name=table_name)
+        res = res.fetchone()[0]
+
+        if re.search(r'(?:CONSTRAINT (\w+) +)?UNIQUE *\({0}\)'.format(col_obj.name), res) is None:
+            return None
```

Comment:
Ah, I see. I guess the difference with nullable is that it defaults to True instead of None? Using None here sounds like the right thing in that case, and an updated comment would help.